modules/pubsub/nats/nats.go

```diff
@@ -0,0 +1,41 @@
+package nats
+
+import (
+	nts "github.com/nats-io/nats.go"
+
+	"github.com/spaceuptech/space-cloud/model"
+)
+
+// nats holds the nats driver session
+type nats struct {
+	conn *nts.EncodedConn
+}
+
+func Connect(connection string) (*nats, error) {
+	nc, err := nts.Connect(connection)
+	c, _ := nts.NewEncodedConn(nc, "json")
+	if err != nil {
+		return nil, err
+	}
+	return &nats{c}, nil
+}
+
+func (n *nats) Publish(subject string, msg *model.PubsubMsg) error {
+	sendChannel := make(chan *model.PubsubMsg)
+	// defer sendChannel.Close()
+	err := n.conn.BindSendChan(subject, sendChannel)
```

Comment:
Why can't we simply publish. We don't really need this here.

Reply:
Without bindingChannels, we can only send a byte array. And this is the ideal way to use nats.
"EncodedConn are the preferred way to interface with NATS"

Reply:
I think marshal that struct and use publish directly. It fits our use case better. This can be a potential memory leak.

Reply:
The interface wont work correctly then. Instead of a generic channel, I will have to use the nats.Msg channel. This wont work properly with other pubsub providers.

Reply:
I think the current thing will not lead to a memory leak, but still. let me research a bit more.

Reply:
Can we simply use this
https://godoc.org/github.com/nats-io/go-nats#EncodedConn.Publish

Reply:
Cool. I'll do that

Reply:
Just noticed, this does not have the queue option.

Reply:
What if I close the channel in defer? We wont have a memory leak then, right?

Reply:
Do you need to specify the queue while publishing? Lets stick to `EncodedConn.Publish()`.

Reply:
Cool